assets/js/uploadFileProgressBar.js

```diff
@@ -0,0 +1,143 @@
+        var progressBar = document.querySelector('.js-progress-bar');
+
+        uploadBtn.classList.add('govuk-!-display-none');
+        progressBar.classList.remove('govuk-!-display-none');
+    },
+
+    pretendToSendFormData: function () {
+        var request = new XMLHttpRequest();
+        var emptyPostRequest = '';
+        var formData = UploadProgressBar.createDataForForm();
+
+        request.upload.addEventListener('progress', function (event) {
+            var progressPct = Math.round((event.loaded / event.total) * 100);
+            var progressBar = document.querySelector('.js-upload-progress-bar');
+
+            progressBar.ariaValueNow = progressPct;
+            progressBar.style.width = progressPct + '%';
+            totalBytesToUpload = event.total;
+            totalBytesUploaded = event.loaded;
+
+            if (event.loaded === event.total) {
+                UploadProgressBar.showFileScanning();
+            }
+        });
+
+        request.onerror = function (err) {
+            console.error(err, 'PretendToSendFormData Error');
+        };
+
+        request.open('post', emptyPostRequest);
+        request.send(formData);
+    },
+
+    createDataForForm: function () {
+        var uploadInput = document.querySelector('.js-multi-file-input');
+        var formData = new FormData();
+        var fileListArr = Array.from(uploadInput.files);
+
+        totalFilesToUpload = uploadInput.files.length;
+        fileListArr.forEach(function (file, index) {
+            formData.append('file' + index, file);
+        });
+
+        return formData;
+    },
+
+    showFileScanning: function () {
+        var progressBar = document.querySelector('.js-upload-progress-bar');
+        var progressBarText = document.querySelector(
+            '.js-upload-progress-text'
+        );
+        var secondsRemaining = document.querySelector(
+            '.js-upload-seconds-remaining'
+        );
+
+        progressBar.classList.add('upload-progress--bar__stripes');
+        progressBarText.innerHTML =
+            'Checking documents & scanning for viruses...';
+        secondsRemaining.style.display = 'none';
+    },
+
+    displayTimeRemaining: function (timeRemainingInSeconds) {
+        var secondsRemainingElem = document.querySelector(
+            '.js-upload-seconds-remaining'
+        );
+        var secondsStr = 'seconds';
+
+        if (timeRemainingInSeconds === 1) {
+            secondsStr = 'second';
+        }
+
+        if (isNaN(timeRemainingInSeconds)) {
+            timeRemainingInSeconds = 0;
+        }
+
+        secondsRemainingElem.innerHTML =
+            timeRemainingInSeconds + ' ' + secondsStr + ' remaining';
+    },
+};
+
+function browserIsIE() {
+    var ua = window.navigator.userAgent;
+    var msie = ua.indexOf('MSIE ');
+    var trident = ua.indexOf('Trident/');
+
+    return msie > 0 || trident > 0;
+}
+
+if (!browserIsIE()) {
+    UploadProgressBar.init();
+
+    /**
+     * Display time remaining
```

Comment:
why time, out of interest? The progress bar seems to be reporting overall progress, eg, uploading, scanning etc, but we're only factoring in the upload part

Reply:
Time is displayed purely because that's what Alvin suggested in his original design:

<img width="916" alt="Screenshot 2022-01-06 at 11 46 19" src="https://user-images.githubusercontent.com/1377253/148378393-344d1c44-5258-45e8-a00a-d7129d6044d3.png">

But you do raise a good point about only showing times for the upload part. It's difficult to predict how long file scanning will take so I didn't factor that in, but it does actually make sense to not display the time remaining 🤔 

I'll raise it with Alvin.

Reply:
We've ended up removing the seconds timer :(